src/components/results/dimension-based-results/dimension/index.js

```diff
@@ -1,63 +1,86 @@
+import PropTypes from "prop-types";
 import {Component} from "react";
+import TraitifyPropType from "lib/helpers/prop-type";
 import withTraitify from "lib/with-traitify";
 import {rgba} from "lib/helpers/color";
 import style from "./style";
 
-class Dimension extends Component{
-  constructor(props){
+class Dimension extends Component {
+  static propTypes = {
+    getOption: PropTypes.func.isRequired,
+    index: PropTypes.number.isRequired,
+    traitify: TraitifyPropType.isRequired,
```

Comment:
Was thinking about whether it would make more sense to present these up front or have them as a reference at the bottom of the component, but I think you're right with having them here